packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -469,4 +479,89 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+      {required final VoidCallback onDoneClick}) {
+    final AppLocalizations localizations = AppLocalizations.of(context);
+    final bool isDark = Theme.of(context).brightness == Brightness.dark;
+
+    return SizedBox(
+      width: double.infinity,
+      child: SmoothCard(
+        clipBehavior: Clip.antiAliasWithSaveLayer,
+        color: Theme.of(context).colorScheme.secondary,
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            Padding(
+              padding: const EdgeInsets.all(SMALL_SPACE),
+              child: Text(
+                _createDoneMessage(localizations),
+                style: Theme.of(context).textTheme.displaySmall,
+              ),
+            ),
+            Row(
+              children: <Widget>[
+                Expanded(
+                  child: Padding(
+                    padding: const EdgeInsets.all(SMALL_SPACE),
+                    child: SmoothSimpleButton(
+                      onPressed: onDoneClick,
+                      buttonColor: Colors.green[700],
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        children: <Widget>[
+                          Flexible(
+                            child: Text(
+                                localizations.new_product_done_button_label),
+                          ),
+                          const SizedBox(width: SMALL_SPACE),
+                          const Icon(Icons.arrow_forward),
+                        ],
+                      ),
+                    ),
+                  ),
+                ),
+                const SizedBox(height: MEDIUM_SPACE),
+                Stack(
+                  alignment: Alignment.bottomRight,
+                  children: <Widget>[
+                    Align(
+                      alignment: Alignment.topLeft,
+                      widthFactor: 0.8,
+                      heightFactor: 0.8,
+                      child: SvgPicture.asset(
+                        isDark
+                            ? 'assets/product/product_completed_graphic_dark.svg'
+                            : 'assets/product/product_completed_graphic_light.svg',
+                        width: 70,
+                      ),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+String _createDoneMessage(AppLocalizations localizations) {
+  final String? name = _getRegisteredUserName();
+  if (name == null) {
+    return localizations.new_product_done_msg_no_user;
+  } else {
+    return localizations.new_product_done_msg(name);
+  }
+}
+
+String? _getRegisteredUserName() {
+  final String? userId = OpenFoodAPIConfiguration.globalUser?.userId;
+  if (userId == null || userId.toLowerCase().contains('welcome')) {
+    return null;
+  } else {
+    return userId;
```

Comment:
Also it would be good to catch the case if `userId.isEmail``and show the default "no user" string